packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -824,6 +848,117 @@ class ThemeData extends Diagnosticable {
     properties.add(new DiagnosticsProperty<IconThemeData>('accentIconTheme', accentIconTheme));
     properties.add(new DiagnosticsProperty<SliderThemeData>('sliderTheme', sliderTheme));
     properties.add(new DiagnosticsProperty<ChipThemeData>('chipTheme', chipTheme));
+    properties.add(new DiagnosticsProperty<AdaptiveWidgetThemeData>('adaptiveWidgetTheme', adaptiveWidgetTheme));
+  }
+}
+
+/// Describes whether descendent Material Design widgets of should automatically
+/// adapt to use Cupertino widgets when running on iOS.
+///
+/// A [Map] from Material widget types to a boolean describes whether the
+/// adaptation should be made for widgets that is wrapped with [PlatformBuilder]
+/// descendant to this [Theme] and whose [PlatformBuilder.themeAdaptiveType]
+/// matches the type in the [adaptiveWidgets] [Map].
+///
+/// When a widget type is not explicitly specified in [adaptiveWidgets], the
+/// [defaultAdaptiveness] value is used.
+///
+/// The singleton [none] instance can be used to signal no adaptation on any
+/// descendent widgets.
+///
+/// The singleton [bundled] instance can be used to signal adaptation on all
+/// Material widgets.
+///
+/// The singleton [all] instance can be used to signal adaptation on all widgets
+/// including user-custom usages of [PlatformBuilder].
+///
+/// Individual [PlatformBuilder] descendents can still override this [Theme]
+/// by specifying null for [PlatformBuilder.themeAdaptiveType].
+@immutable
+class AdaptiveWidgetThemeData extends Diagnosticable{
+  /// Create a [AdaptiveWidgetThemeData] for a [Theme] that instructs whether
+  /// descendent widgets should adapt to the platform.
+  ///
+  /// The parameters [adaptiveWidgets] and [defaultAdaptiveness] cannot be null.
+  const AdaptiveWidgetThemeData(
+    this.adaptiveWidgets, {
+    this.defaultAdaptiveness,
+  }) : assert(adaptiveWidgets != null);
+
+  /// A [AdaptiveWidgetThemeData] where none of the descendents dynamically
+  /// adapt to the platform unless overridden in [PlatformBuilder].
+  static const AdaptiveWidgetThemeData none =
+      const AdaptiveWidgetThemeData(const <Type, bool>{});
+
+  /// A [AdaptiveWidgetThemeData] where all bundled Material Design widgets
+  /// that support platform adaptations will adapt when running on iOS.
+  static const AdaptiveWidgetThemeData bundled =
+      const AdaptiveWidgetThemeData(const <Type, bool>{
+        FlatButton: true,
+        RaisedButton: true,
+        Slider: true,
```

Comment:
shouldn't this match the material spec list?

Reply:
I don't know that there's an explicit list. Maybe @willlarche has one. The material site just has a screenshot and I don't think it's all-inclusive.

Reply:
specifically, looks like radio buttons and switches should adapt, and nothing else, in the "recommended" set.

Reply:
based on https://material.io/design/platform-guidance/cross-platform-adaptation.html#cross-platform-guidelines

Reply:
I think it's a question for @willlarche.

Based on https://github.com/flutter/flutter/pull/17675#issuecomment-391125670 and below, it seems like it's just an example of what's possible rather than a set of guidelines.

Reply:
I'm not satisfied with the guidance here. I'd like to press them for more specificity. 

But for now, _all_ selection controls should adapt: slider should be `UISlider`, radio buttons and check boxes should be `UIButtons` with a check icon for `image forState:UIControlStateSelected`.